test/CMakeLists.txt

```diff
@@ -56,21 +56,23 @@ if (SINGULARITY_BUILD_CLOSURE)
 endif()
 
 if (SINGULARITY_USE_HDF5 AND SINGULARITY_TEST_STELLAR_COLLAPSE)
+  set(SINGULARITY_GOLDFILE_URL
+    https://github.com/lanl/singularity-eos/releases/download/${SINGULARITY_GOLDFILES_VERSION}/goldfiles.tar.gz)
+  message(STATUS "Attempting to download gold files for regression tests. "
+    "File is located at ${SINGULARITY_GOLDFILE_URL} "
+    "If you do not want to use the gold file, you can set -DSINGULARITY_TEST_STELLAR_COLLAPSE=OFF."
+    )
+  include(FetchContent)
+  FetchContent_Declare(goldfiles
+    URL ${SINGULARITY_GOLDFILE_URL}
+    URL_HASH SHA256=${SINGULARITY_GOLDFILE_HASH}
+    SOURCE_DIR goldfiles
+    )
+  FetchContent_MakeAvailable(goldfiles)
```

Comment:
With `cmake@>=3.24`, you'll get this warning blurb.

```bash
[cmake] CMake Warning (dev) at /usr/share/cmake/Modules/FetchContent.cmake:1267 (message):
[cmake]   The DOWNLOAD_EXTRACT_TIMESTAMP option was not given and policy CMP0135 is
[cmake]   not set.  The policy's OLD behavior will be used.  When using a URL
[cmake]   download, the timestamps of extracted files should preferably be that of
[cmake]   the time of extraction, otherwise code that depends on the extracted
[cmake]   contents might not be rebuilt if the URL changes.  The OLD behavior
[cmake]   preserves the timestamps from the archive instead, but this is usually not
[cmake]   what you want.  Update your project to the NEW behavior or specify the
[cmake]   DOWNLOAD_EXTRACT_TIMESTAMP option with a value of true to avoid this
[cmake]   robustness issue.
```

See https://cmake.org/cmake/help/latest/policy/CMP0135.html

Not breaking, and for this case I think is over-doing it. But if you want to avoid terminal bloat, you could add 

```cmake
# Avoid warning about DOWNLOAD_EXTRACT_TIMESTAMP in CMake 3.24:
if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.24.0")
  cmake_policy(SET CMP0135 NEW)
endif()
```